cql3/expr/prepare_expr.cc

```diff
@@ -1239,6 +1239,20 @@ binary_operator prepare_binary_operator(binary_operator binop, data_dictionary::
     lw_shared_ptr<column_specification> rhs_receiver = get_rhs_receiver(lhs_receiver, binop.op);
     expression prepared_rhs = prepare_expression(binop.rhs, db, table_schema.ks_name(), &table_schema, rhs_receiver);
 
+    // IS NOT NULL requires an additional check that the RHS is NULL.
+    // Otherwise things like `int_col IS NOT 123` would be allowed - the types match, but the value is wrong.
+    if (binop.op == oper_t::IS_NOT) {
+        bool rhs_is_null = is<constant>(prepared_rhs) && as<constant>(prepared_rhs).is_null();
+
+        if (!rhs_is_null) {
+            expression binop_expr = binop;
```

Comment:
nitpick: why do we need this extra copy of this variable?

Reply:
The printer keeps a `expression&`, but `binop` has type `binary_operator`, so I have to create an instance of `expression` to get the reference.